internal/targets/delegation.go

```diff
@@ -18,13 +19,23 @@ type delegationsIterator struct {
 }
 
 // NewDelegationsIterator initialises an iterator with a first step
-// on top level targets
-func NewDelegationsIterator(target string) *delegationsIterator {
+// on top level targets.
+func NewDelegationsIterator(target string, topLevelKeysDB *verify.DB) *delegationsIterator {
+	role := topLevelKeysDB.GetRole("targets")
+	keyIDs := []string{}
+	if role != nil {
```

Comment:
Would it be an error or do we never expect that the keys DB never have targets?

Reply:
I made this return an error if the top level DB is missing the targets role, and added tests to exercise this functionality.